### ISSUE TITLE
Possible bug (duplicated responses)

### ISSUE THREAD

**Issue by MaRmARk0** [knuckleswtf/scribe#601]
I ran into an issue where I generate custom structure in my BaseController::inheritedDocsOverrides(). It's working fine until when I return multiple `response` arrays, like so:

```
... 
"responses" => array:2 [
    0 => array:2 [
      "status" => 200
      "content" => """
        {\n
            "id": 5,\n
            "group": "text.testing",\n
            "key": "some_key_3",\n
            "locale": "sk-SK",\n
            "text": "Dolor sit amet in sk-SK",\n
            "description": "This is internal description of given translation.",\n
            "is_internal": false,\n
            "created_at": "2022-01-01 11:00:00",\n
            "updated_at": null\n
        }\n
        """
    ]
    1 => array:2 [
      "status" => 422
      "content" => """
        {\n
            "message": "Validation errors",\n
            "errors": {\n
                "some_field_1": [\n
                    "Error message 1",\n
                    "Error message 2",\n
                    "Error message N"\n
                ],\n
                "some_field_2": [\n
                    "Error message 1"\n
                ]\n
            }\n
        }\n
        \n
        """
    ]
...
```
Unfortunately after calling scribe:generate I get doubled the count of responses:
![This is an image](https://imgur.com/rPj6HuO.png)

If my `responses` array contains 3 responses, resulting docs contains 6 responses and so on.
I suspect `Extractor::processRoute()` which is calling 
```
        $this->fetchResponses($endpointData, $routeRules);
        $this->mergeInheritedMethodsData('responses', $endpointData, $inheritedDocsOverrides);
```
which is doing
```
$endpointData->responses->concat($overrides);
```
and that line duplicates every responses.

Note that I have disabled all strategies except this one:
```
        'metadata' => [
            Strategies\Metadata\GetFromDocBlocks::class,
        ],
```
because we want to solely use `inheritedDocsOverrides()` in `BaseController`. We are *not* using doc-blocks anywhere - GetFromDocBlocks::class is there just for sake of generating navigation in final docs.

**My environment:**
 - PHP 8.1.12 (cli)
 - Laravel 9.38
 - Scribe 4.11

**Comment by shalvah** [comment]
Hmm, so, if you remove the `concat` line, are responses no longer duplicated?

**Comment by MaRmARk0** [comment]
If I removed `concat($overrides)` completely, no responses were generated at all. 

But if I changed it like this

```
        if (is_array($overrides)) {
            foreach ($overrides as $key => $item) {
                switch ($stage) {
                    case "responses":
                        // concat only single $item per iteration, not full $overrides array
                        $endpointData->responses->concat([$item]); 
                        $endpointData->responses->sortBy('status');
                        break;
```

it seems to work on all endpoints. :)

**Comment by shalvah** [comment]
Can you try again now? I think this may have been fixed by #623
